lib/emulator.cpp

```diff
@@ -38,25 +38,23 @@ using namespace irve::internal;
 emulator::emulator_t::emulator_t(int imagec, const char* const* imagev):
     m_CSR(),
     m_memory(imagec, imagev, m_CSR),
-    m_cpu_state(m_CSR),
+    m_cpu_state(),
     m_intercept_breakpoints(false)
 {
     irvelog(0, "Created new emulator instance");
 }
 
 bool emulator::emulator_t::tick() {
-    this->m_CSR.implicit_write(Csr::Address::MINSTRET, this->m_CSR.implicit_read(Csr::Address::MINSTRET) + 1);
-    this->m_CSR.implicit_write(Csr::Address::MCYCLE,   this->m_CSR.implicit_read(Csr::Address::MCYCLE  ) + 1);
+    this->m_CSR.increment_perf_counters();
     irvelog(0, "Tick %lu begins", this->get_inst_count());
 
     //Any of these could lead to exceptions (ex. faults, illegal instructions, etc.)
     try {
         decode::DecodedInst decoded_inst = this->fetch_and_decode();
         this->execute(decoded_inst);
     } catch (const rv_trap::RvException& e) {
-        uint32_t raw_cause = (uint32_t)e.cause();
-        assert((raw_cause < 32) && "Unsuppored cause value!");//Makes it simpler since this means we must check medeleg always
-        irvelog(1, "Handling exception: Cause: %u", raw_cause);
+        assert((uint32_t)e.cause() && "Unsuppored cause value!");//Makes it simpler since this means we must check medeleg always
```

Comment:
Wow, yeah good catch. I did not mean to change this really at all.

The issue was that with NDEBUG set and without logging this would error-out because the `raw_cause` variable would go unused.

I _meant_ to just manually inline `(uint32_t)e.cause()` into both statements, but I really goofed there it would seem.

I will change this to what I meant to do:
```
assert((((uint32_t)e.cause()) < 32) && "Unsuppored cause value!");//Makes it simpler since this means we must check medeleg always
irvelog(1, "Handling exception: Cause: %u", (uint32_t)e.cause());
```

Reply:
I though I had removed the `irvelog` line too but it looks like it was still there, so this wasn't as bad as I though. Regardless it's fixed!